src/Core/Resettable.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace SilverStripe\Core;
+
+/**
+ * Represents a class with a local cache which normally invalidates itself between requests.
+ *
+ * Designed so that tests can automatically flush these objects between runs in lieu
+ * of a real change.
+ */
+interface Resettable
```

Comment:
My point is that introducing *more* use cases is taking it in the wrong direction. But you're right, this is best addressed separately - the interface itself is good, we might just move all those to explicit registration at some point.